code/modules/mob/living/carbon/human/species/unathi.dm

```diff
@@ -131,12 +131,15 @@
 	var/datum/action/innate/ignite/ash_walker/fire = new()
 	fire.Grant(H)
 	RegisterSignal(H, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(speedylegs))
+	if(!is_mining_level(H.z))
```

Comment:
Can you just call the `speedylegs` proc here?

Reply:
No, speedylegs only gets added or removed on z-level transition, so calling speedylegs there will not do anything, if the egg is on station, or a cling transforms on station.

Reply:
The signal makes it so that it gets *called* on z-level transition, that does not stop us from calling it here manually.

Reply:
Just call `speedylegs()`